sensenet/layers/tree.py

```diff
@@ -58,38 +55,39 @@ def call(self, inputs):
 
         return tf.gather(self._outputs, tf.reshape(out_idxs, (-1,)))
 
-class DecisionForest(tf.keras.layers.Layer):
-    def __init__(self, trees):
-        super(DecisionForest, self).__init__()
-
+class DecisionForest():
+    def __init__(self, **kwargs):
+        self._forest = kwargs['forest']
         self._trees = []
 
-        for tree in trees:
+        for tree in self._forest:
             if len(tree) == 2 and tree[-1] is None:
-                self._trees.append(DecisionNode(tree))
+                self._trees.append(DecisionNode(tree=tree))
             else:
-                self._trees.append(DecisionTree(tree))
+                self._trees.append(DecisionTree(tree=tree))
 
-    def call(self, inputs):
+    def __call__(self, inputs):
         all_preds = []
 
         for tree in self._trees:
             preds = tree(inputs)
             all_preds.append(preds)
 
-        summed = tf.math.accumulate_n(all_preds)
+        summed = tf.math.add_n(all_preds)
 
         return summed / len(all_preds)
 
 class ForestPreprocessor(tf.keras.layers.Layer):
```

Comment:
> Why do we have Forests and Trees in sensenet?

This is because we have `tree_embedding` in deepnets, which is basically just a tree-based convolution that improves performance on a handful of datasets.  It's kind of a strange thing for this library to have and in retrospect I probably shouldn't have done it.

On the plus side, supporting it meant I had to develop a tree-based custom operator for TensorFlow.  In my tests, when I use this operator to instantiate a random forest predictor, predictions are about 20-30% faster than scikit-learn on the same forest.  So if we ever need our random forests to go crazy fast for some reason, we can always run them through TensorFlow.